2025REDOBuildSeason/src/main/java/frc/robot/Auto/AutoMissionExecutor.java

```diff
@@ -0,0 +1,77 @@
+
+package frc.robot.Auto;
+
+import frc.robot.Auto.Missions.MissionBase;
+
+/**
+ * This class selects, runs, and (if necessary) stops a specified autonomous Mission.
+ */
+public class AutoMissionExecutor {
+    private MissionBase mAutoMission = null;
+    private Thread mThread = null;
+
+    public void setAutoMission(MissionBase new_auto_Mission) {
+        mAutoMission = new_auto_Mission;
+        mThread = new Thread(new Runnable() {
+            
+            @Override
+            public void run() {
+                if (mAutoMission != null) {
+                    mAutoMission.run();
+                }
+            }
+        });
+    }
+
+    public void start() {
+        if (mThread != null) {
+            //System.out.println("mThread is not null");
+            mThread.start();
+        }
+    }
+
+    public boolean isStarted() {
+        return mAutoMission != null && mAutoMission.isActive() && mThread != null && mThread.isAlive();
+    }
+
+    public void reset() {
+        if (isStarted()) {
+            stop();
+        }
+
+        mAutoMission = null;
+    }
+
+    public void stop() {
+        if (mAutoMission != null) {
+            mAutoMission.stop();
+        }
+
+        mThread = null;
+    }
+
+    public MissionBase getAutoMission() {
+        return mAutoMission;
+    }
+
+    public boolean isInterrupted() {
+        if (mAutoMission == null) {
```

Comment:
Could do this whole function as 
```suggestion
        return mAutoMission == null ? false :  mAutoMission.getIsInterrupted();
```